### PR TITLE
week1_day3

### PR DESCRIPTION
**Pull request by yashraj-01**
Exercise: [here](https://docs.google.com/document/d/193B2OXkQIJ2Va1JjxZ9iVDYKvaF1TwRQcaDqkg-HMzA/edit#heading=h.tytft2pkn5l9)

The project structure is inspired by this repo: https://github.com/speakeasy-api/rest-template-go/

Below screenshots show working api requests:

Terminal:
Screenshot 2023-10-17 at 1 14 33 PM

Postman:
Screenshot 2023-10-17 at 1 24 14 PM

### CODE REVIEW

File: week1_day3/exercise1/Config/Database.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+// DBConfig represents db configuration
+type DBConfig struct {
+	Host     string
+	Port     int
+	User     string
+	DBName   string
+	Password string
+}
+
+func BuildDBConfig() *DBConfig {
```
Comment (lokpati-rzp) [inline week1_day3/exercise1/Config/Database.go]: Only New

File: week1_day3/exercise1/Config/Database.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		Host:     "localhost",
+		Port:     3306,
+		User:     "root",
+		Password: "yashDBraj",
+		DBName:   "exercise1",
+	}
+	return &dbConfig
+}
+
+func DbURL(dbConfig *DBConfig) string {
```
Comment (lokpati-rzp) [inline week1_day3/exercise1/Config/Database.go]: make it unexported.

File: week1_day3/exercise1/Config/Database.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package Config
```
Comment (lokpati-rzp) [inline week1_day3/exercise1/Config/Database.go]: use this https://github.com/speakeasy-api/rest-template-go/ and structure your code accordingly.

File: week1_day3/exercise1/Config/Database.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+import (
+	"fmt"
+	"github.com/jinzhu/gorm"
+)
+
+var DB *gorm.DB
```
Comment (lokpati-rzp) [inline week1_day3/exercise1/Config/Database.go]: avoid using singletons.

File: week1_day3/exercise1/Models/User.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+// CreateUser ... Insert New data
+func CreateUser(user *User) (err error) {
+	if err = Config.DB.Create(user).Error; err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetUserByID ... Fetch only one user by Id
+func GetUserByID(user *User, id string) (err error) {
```
Comment (lokpati-rzp) [inline week1_day3/exercise1/Models/User.go]: take only id as a input and return (*User,error)

File: week1_day3/exercise1/Models/User.go [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+// UpdateUser ... Update user
+func UpdateUser(user *User, id string) (err error) {
+	fmt.Println(user)
+	Config.DB.Save(user)
+	return nil
+}
+
+// DeleteUser ... Delete user
+func DeleteUser(user *User, id string) (err error) {
```
Comment (lokpati-rzp) [inline week1_day3/exercise1/Models/User.go]: DeleteUserByID and return interface should be (User,error)

**Review by lokpati-rzp — COMMENTED** [review]
https://github.com/speakeasy-api/rest-template-go/ 
Use the above link to structure your code.

**Comment by yashraj-01** [comment]
@lokpati-rzp I've restructured the project. PTAL. I believe most your comments are addressed in it except using `context.Context`. I'm still using `gin.Context` as I don't fully understand how to use `context.Context`.

**Review by lokpati-rzp — APPROVED** [review]

**Pull request merged**